### PR TITLE
Workaround for https://github.com/dotnet/roslyn/issues/46859 

### PR DESCRIPTION
**Pull request by mavasani**
**Strongly recommended to review ignoring white space diffs: https://github.com/dotnet/roslyn-analyzers/pull/4026/files?diff=unified&w=1**

See https://github.com/dotnet/roslyn/issues/46859 and https://github.com/dotnet/roslyn-analyzers/pull/3917#issuecomment-674477914 for details. We might end up with corrup pooled data structures in presence of cancellation, so work around the Roslyn bug by not returning pooled object back to the pool in presence of cancellation.

### CODE REVIEW

File: src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/PointsToAnalysis/PointsToAnalysis.PointsToDataFlowOperationVisitor.cs [hunk 1]
```diff
@@ -4,6 +4,7 @@
 using System.Collections.Generic;
 using System.Collections.Immutable;
 using System.Diagnostics;
+using System.Threading;
```
Comment (mavasani) [inline src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/PointsToAnalysis/PointsToAnalysis.PointsToDataFlowOperationVisitor.cs:7]:
```suggestion
```

**Comment by dotpaul** [comment]
Could this be done in 2.9.x?

**Comment by mavasani** [comment]
I am planning to actually fix the analyzer driver bug and close this PR out. Even after these changes, we see similar asserts cropping up.

**Comment by dotpaul** [comment]
Ah okay. I guess this issue is mostly about the IDE, right? So if people run into this, the solution will be to upgrade their Visual Studio?

**Comment by mavasani** [comment]
Yes, no build impact. Only affects IDE typing scenario.

**Comment by mavasani** [comment]
Actually, turns out this is not an analyzer driver bug. Analyzer is expected to not throw cancellation exception or free resources on potentially cancellable code as driver will make the same callback next time. I am going to close this PR and open one targeting 2.9.x

**Pull request closed without merging**